addons/website_sale/controllers/main.py

```diff
@@ -1034,7 +1034,7 @@ def checkout_form_validate(self, mode, all_form_values, data):
             error_message.append(_('Invalid Email! Please enter a valid email address.'))
 
         # vat validation
-        Partner = request.env['res.partner']
+        Partner = request.env['res.partner'].sudo()
```

Comment:
Could you please restrict the use of `sudo` to the necessary scope ?
`partner_dummy.sudo().check_vat()`